go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -277,9 +277,11 @@ func (root *RootValue) GenerateTagsForNewColumns(ctx context.Context, tableName
 	}
 
 	newTags := make([]uint64, len(newColNames))
+	existingTags := set.NewUint64Set(rootSuperSchema.AllTags())
 	for i := range newTags {
-		newTags[i] = schema.AutoGenerateTag(rootSuperSchema, tableName, existingColKinds, newColNames[i], newColKinds[i])
+		newTags[i] = schema.AutoGenerateTag(existingTags, tableName, existingColKinds, newColNames[i], newColKinds[i])
 		existingColKinds = append(existingColKinds, newColKinds[i])
+		existingTags.Add(newTags[i])
```

Comment:
@bheni this is why `irs-soi` was failing. It had so many columns that it was creating tag collisions while generating tags for a single table. Updating the existing tag set on each tag creation fixes it.